packages/core/strapi/src/services/entity-validator/blocks-validator.ts

```diff
@@ -1,7 +1,7 @@
 import { yup } from '@strapi/utils';
 
 const textNodeValidator = yup.object().shape({
-  type: yup.string().equals(['text']).required(),
```

Comment:
I don't understand, why is it no longer required. It doesn't match the schema we defined in the RFC

Reply:
Slate adds blocks with this structure `{ text: 'hello' }` without type. These are leafs and (in theory) they are not inline nodes.

One of the rules of Slate said: You can't have an inline node as first/last child of a block, so when you have a block which is only a link, or a link at the beginning/end, Slate automatically will add this without type node. (You can read more about this in the PR description).

I know this doesn't match our schema, but the only way I found to avoid slate doing this is replacing completely the normalize function, which is possible but I don't think makes too much sense

Reply:
But couldn't we create a new validator specific for text specific to links. That way it's only an exception for links and we can still be strict with the other nodes.

Reply:
@markkaylor  Other option is extend the `normalizeNode` from Slate to add `type: text` to all the text nodes. If you see [here](https://docs.slatejs.org/api/nodes/text) for them a Text node don't have the need to have a type field, we are adding it to match our schema, so makes sense we make sure we always have a type by extending the function

Reply:
Nice! Yeah that sounds like a good idea.